README.md

```diff
@@ -93,10 +75,23 @@ Once you have Carthage [installed](#installing-carthage), you can begin adding f
 
 ### Getting started
 
+#### Building platform-independent xcframeworks (Xcode 12 and above)
+
+1. Create a [Cartfile][] that lists the frameworks you’d like to use in your project.
+1. Run `carthage update`. This will fetch dependencies into a [Carthage/Checkouts][] folder and build each one or download a pre-compiled xcframework.
+1. On your application targets’ _General_ settings tab, in the _Embedded Binaries_ section, drag and drop each xcframework you want to use from the [Carthage/Build][] folder on disk.
+
+#### Building platform-specific framework bundles (default for Xcode 11 and below)
+
+**Xcode 12+ incompatibility**: Multi-architecture platforms are not supported when building framework bundles in Xcode 12 and above. Prefer [building with xcframeworks](#building-platform-independent-xcframeworks-xcode-12-and-above). If you need to build discrete framework bundles, [use a workaround xcconfig file](Documentation/Xcode12Workaround.md).
+
 ##### If you're building for macOS
 
+<details>
+	<summary>macOS-specific instructions</summary>
+
 1. Create a [Cartfile][] that lists the frameworks you’d like to use in your project.
-1. Run `carthage update --platform macOS`. This will fetch dependencies into a [Carthage/Checkouts][] folder and build each one or download a pre-compiled framework.
+1. Run `carthage update --platform macOS --no-create-xcframework`. This will fetch dependencies into a [Carthage/Checkouts][] folder and build each one or download a pre-compiled framework.
```

Comment:
isn't the build argument false by default?

Reply:
It depends on the current Xcode version. If you're on 12, `--create-xcframework` is default, if you're on 11 or below, `--no-create-xcframework` is default: https://github.com/Carthage/Carthage/pull/3071/files#diff-1ef6915d12e67971af0a706b0f50605779bf84b0bb3593e90b0862a4f01b64af

So this section of the readme is written to be correct regardless of your Xcode.

tbh, I don't like how dynamic this is, but it feels like the right compromise. If the argument was false by default, this PR breaks backwards-compatibility for users on older Xcodes, _or_ it requires devs on Xcode 12 to always pass `--create-xcframework` to get a working build.

Reply:
I wouldn't expect the behaviour of the default flags to change depending on external conditions (i.e. the installed version of Xcode). Imho it should always be `--no-create-xcframeworks` unless otherwise specified.

Reply:
this makes it very hard to predict what carthage will do just by looking at the CLI. Think of scrips and other tooling (bug reports) that would also have to independently check for the installed version of Xcode.

Reply:
Totally understand the concern. I'm willing to make --no-create-xcframeworks the default. I should be able to add an error when Carthage can anticipate that lipo will fail, so that users know they need to re-reun with --create-xcframeworks.

Reply:
Why keep a default that is broken on the latest version of Xcode? It will only lead to increased support burden as confused users don't understand why Carthage is still broken on Xcode 12 and requires looking at the documentation to discover the new special flag `--create-xcframeworks`. 

There are a lot of developers incorrectly removing ARM iOS Simulator slices to work around the Carthage Xcode 12 issues. There is no future workflow where the old behavior of `--no-create-xcframeworks` is correct: users of Apple Silicon Macs need xcframeworks for the ARM iOS Simulator, and as well as those who are shipping Mac Catalyst apps.

I would strongly urge to use defaults that "just work".

Reply:
To further support the above, Xcode 12 will be required by Apple for all iOS and iPadOS apps in just a few months, and I suspect the vast majority of developers will upgrade (even if they only develop macOS apps), so the default working only in old versions doesn't seem very useful.

Reply:
Yeah @chrisballinger, I think that's the strongest argument to defaulting to true. It's also what I had in mind when I implemented this, and I agree that it's a "bad default" from the perspective of an Xcode 12 user.

If we take @tmspzz's suggestion and do the conservative thing, I think we _must_ give users a meaningful error directing them to the argument.

Reply:
@elliottwilliams Why not just always build xcframeworks on both Xcode 11 and Xcode 12? That would get rid of the ambiguity between versions, and will always do the correct thing going forward. Users are going to have to deal with a breaking change at some point, and it might as well be now. I guess Xcode 11 doesn't support the `-debug-symbols` flag for dSYMs/bcsymbolmaps in xcframeworks, but perhaps that could be guarded against.

Reply:
> There is no future workflow where the old behavior of --no-create-xcframeworks is correct: users of Apple Silicon Macs need xcframeworks for the ARM iOS Simulator, and as well as those who are shipping Mac Catalyst apps.
> I would strongly urge to use defaults that "just work".

Carthage is compatible all the way down to High Sierra.

A new flag should not be enabled by default.

Reply:
A middle ground would be ending support for all systems where Xcode 12 is not supported before cutting the release with support for XCFrameworks.

The defaulting to `--create-xcframeworks` would make sense

Reply:
> A new flag should not be enabled by default.

Also to consider, certain CI systems that upgrade Carthage without user confirmation. If avoidable, we aim to provide least-possible disruption to those users (continuing to compile their projects is ideal).

Reply:
> A middle ground would be ending support for all systems where Xcode 12 is not supported before cutting the release with support for XCFrameworks.
>
> Then defaulting to `--create-xcframeworks` would make sense.

If you all are open to this, I think it's a good direction to move in. It would also give us an opportunity to drop source compatibility with Xcode 10.1 / Swift 4 and modernize some of Carthage's dependencies.

But I don't want this PR to hinge on the larger task of dropping support for older OSes and Carthage versions. And I'd like to avoid bikeshedding on the default value of this argument, as much as possible 😆

So how about: we do the safe thing and default to --no-create-xcframeworks, _with the understanding that --create-xcframeworks will become the default in the future_. Once this is reviewed and merged, dropping support for older Xcodes is one of the things I / others can start to work on. We can safely make a release with these changes before then.

Reply:
This is done: https://github.com/Carthage/Carthage/pull/3071/commits/9b61e11133810e0ce0ab4454187389cc9457285d

One thing I was thinking about is that we _could_ make --create-xcframeworks the default _if and only if_ Carthage knows that it's going to encounter a lipo failure. This would preserve the behavior of any currently-successful carthage invocation, including when the user is using the EXCLUDED_ARCHS workaround. But for iOS devs building on Xcode 12+ (which I imagine is 90%+ of our users), they would always get xcframeworks instead of build errors.

Not sure I want to do this now, because it's additive and it would require refactoring the point in the build where we decide whether to make an xcframework or a universal framework. Currently we do this immediately after building a scheme, so we'd instead have to wait until all schemes are built to decide how to merge their products.